### PR TITLE
Implement TAR-based directory archiver

### PR DESCRIPTION
**Pull request by elad335**
* Implement the ability to save directory contents as TAR.
  * With the ability to customize saving of specific files and also select which files are saved and which aren't. 
* Implement full original TAR USTAR 255 characters path support.
* Implement file modification time and access time support for extracted/archived TAR.

This will be used for savestates support, optionally saving /dev_hdd1 and game data directories for when "Suspend Emulation Savestate Mode" is disabled.

### CODE REVIEW

File: rpcs3/Loader/TAR.cpp [hunk 1]
```diff
@@ -189,6 +222,14 @@ bool tar_object::extract(std::string vfs_mp)
 			if (file)
 			{
 				file.write(static_cast<fs::container_stream<std::vector<u8>>*>(data.get())->obj);
+				file.close();
+
+				if (mtime != umax && !fs::utime(result, atime, mtime))
+				{
+					tar_log.error("TAR Loader: fs::utime filed on %s (%s)", result, fs::g_tls_error);
```
Comment (Nekotekina) [inline rpcs3/Loader/TAR.cpp:229]: Typo "filed"

File: rpcs3/Loader/TAR.cpp [hunk 2]
```diff
@@ -217,11 +264,95 @@ bool tar_object::extract(std::string vfs_mp)
...
+			*ptr = static_cast<char>('0' + (i % 8));
+		}
+	};
+
+	TARHeader header{};
+	std::memcpy(header.magic, "ustar3", 6);
+
+	const std::string_view saved_path{target_path.data() + src_dir.size(), target_path.size() - src_dir.size()};
+
+	// Prefer saving to name feild as much as we can
```
Comment (Nekotekina) [inline rpcs3/Loader/TAR.cpp:318]: Maybe it's better to follow existing standards for header extensions, in order to keep the tarball readable by existing tools.

Comment (ArchSonix) [reply rpcs3/Loader/TAR.cpp:318]:
```suggestion
	// Prefer saving to name field as much as we can
```

**Comment by elad335** [comment]
Removed extended paths support, windows does not support them without explicit override which wasn't an issue in the past.

File: rpcs3/Loader/TAR.cpp (hunk 1 above)
Comment (elad335) [reply rpcs3/Loader/TAR.cpp:218]: Fixed.

**Comment by elad335** [comment]
* Added the ability to customize saving of specific files and also select which files are saved and which aren't. By using an optional function resposible for saving the file.

**Review by kd-11 — APPROVED** [review]
lgtm

**Pull request merged**